Lib/fontTools/pens/hashPointPen.py

```diff
@@ -0,0 +1,58 @@
+# Modified from https://github.com/adobe-type-tools/psautohint/blob/08b346865710ed3c172f1eb581d6ef243b203f99/python/psautohint/ufoFont.py#L800-L838
+import hashlib
+
+from fontTools.misc.transform import Identity
+from fontTools.pens.pointPen import AbstractPointPen
+
+
+class HashPointPen(AbstractPointPen):
+    """
+    This pen can be used to check if a glyph's contents (outlines plus
+    components) have changed.
+
+    Components are added as the original outline plus each composite's
+    transformation.
+    """
+
+    DEFAULT_TRANSFORM = Identity
+
+    def __init__(self, glyph, glyphSet=None):
```

Comment:
I find a bit strange that we pass glyph in the constructor (just to get its width) but then we require one to call glyph.drawPoints(pen).
I think we should only pass the width to the pen constructor, not the entire glyph.

Reply:
I have changed it.